biz.aQute.remote/src/aQute/remote/util/AgentSupervisor.java

```diff
@@ -45,25 +46,28 @@ protected void connect(Class<Agent> agent, Supervisor supervisor, String host, i
 		connect(agent, supervisor, host, port, -1);
 	}
 
-	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port, int timeout)
+	protected void connect(Class<Agent> agent, Supervisor supervisor, String host, int port, final int timeout)
```

Comment:
I am not sure what the point of connectWait is and retrying. Since the socket wait is the same as the requested wait, when will retry - connect wait ever be greater than zero?

This seems to have gotten way more complicated.